src/Package/Impl/Debugger/DataTips/DataTipTextViewConnectionListener.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Collections.ObjectModel;
+using System.ComponentModel.Composition;
+using Microsoft.R.Components.ContentTypes;
+using Microsoft.VisualStudio.R.Package.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.VisualStudio.R.Package.Debugger.DataTips {
+    [Export(typeof(IWpfTextViewConnectionListener))]
+    [ContentType(RContentTypeDefinition.ContentType)]
+    [TextViewRole(PredefinedTextViewRoles.Debuggable)]
+    internal sealed class DataTipTextViewConnectionListener : IWpfTextViewConnectionListener {
+        public DataTipTextViewConnectionListener() {
+        }
+
+        public void SubjectBuffersConnected(IWpfTextView textView, ConnectionReason reason, Collection<ITextBuffer> subjectBuffers) {
+            var debugger = VsAppShell.Current.GetGlobalService<IVsDebugger>();
+            new DataTipTextViewFilter(textView, debugger);
+        }
+
+        public void SubjectBuffersDisconnected(IWpfTextView textView, ConnectionReason reason, Collection<ITextBuffer> subjectBuffers) {
```

Comment:
It may be necessary to remove it explicitly with IVsTextView.RemoveCommandFilter depending on the particular view lifetime